src/SWP/Bundle/ContentBundle/Loader/ArticleLoader.php

```diff
@@ -125,7 +126,34 @@ public function load($type, $parameters, $responseType = LoaderInterface::SINGLE
                 ));
 
                 if ($route) {
-                    $articles = $this->dm->getReferrers($route, null, null, null, 'SWP\Bundle\ContentBundle\Doctrine\ODM\PHPCR\Article');
+                    $node = $this->dm->getNodeForDocument($route);
+                    $identifier = $node->getIdentifier();
+
+                    $queryStr = 'SELECT * FROM [nt:unstructured] as S ';
+                    $queryStr .= "WHERE S.phpcr:class='SWP\Bundle\ContentBundle\Doctrine\ODM\PHPCR\Article' ";
+                    $queryStr .= "AND S.route=$identifier";
```

Comment:
And it think that whole discussion here is not needed as You should use `setParameter/setParameters` on `Query` object. 
```
$query = $this->dm->createPhpcrQuery($queryStr, QueryInterface::JCR_SQL2);
$query->setParameter('identifier', $identifier);
```